### PR TITLE
#14 module loader instanciation

### PR DESCRIPTION
**Pull request by florian-vuillemot**

### CODE REVIEW

File: src/moduleLoader/ModuleLoader.hpp [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  }
+
+private:
+#ifdef __linux__
+  void *sdl_library = nullptr;
+#elif _WIN32
+  HMODULE sdl_library = nullptr;
+#endif
+  /**
+   * Has we manage this pointer on the class, we didn't use a smart pointer.
```
Comment (MickaToast) [inline src/moduleLoader/ModuleLoader.hpp]: As we

File: src/moduleLoader/ModuleLoader.hpp [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+   */
+  std::string const externCreateObject = "create_object";
+  /**
+   * Mandatory function that delete a T*
+   */
+   std::string const externDestructorObject = "destroy_object";
+  
+public:
+  DynamicLink() = default;
+  ~DynamicLink(){
```
[the remark this answers is not in the document]
Comment (florian-vuillemot) [reply src/moduleLoader/ModuleLoader.hpp]: Ok, as you want

File: src/moduleLoader/exemple/main.cpp [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#include "DynamicLink.hpp"
```
[the remark this answers is not in the document]
Comment (florian-vuillemot) [reply src/moduleLoader/exemple/main.cpp]: I think is not really import, it's mainly a compile time test.
But if of course I can do it.

File: src/moduleLoader/ModuleLoader.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (florian-vuillemot) [reply src/moduleLoader/ModuleLoader.hpp]: I can return a object like it: std::tuple<std::shared<>, std::function<>> with the function as destructor?

[the remark this answers is not in the document]
Comment (polosskevin) [reply src/moduleLoader/ModuleLoader.hpp]: you can build a shared pointer while giving the destructor: https://stackoverflow.com/questions/12340810/using-custom-deleter-with-stdshared-ptr

**Pull request closed without merging**